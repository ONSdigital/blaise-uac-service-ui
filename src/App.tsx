import React, { ReactElement} from "react";
import { Switch, Route, Link } from "react-router-dom";
import UploadSamplePage from "./components/UploadSample/UploadSamplePage";
import ImportUacPage from "./components/ImportUac/ImportUacPage";
import {Footer, Header, BetaBanner, NotProductionWarning} from "blaise-design-system-react-components";
import "./style.css";
import InstrumentListPage from "./components/InstrumentList/InstrumentListPage";

const divStyle = {
    minHeight: "calc(67vh)"
};

function App(): ReactElement {

    return (
        <>
            {
                (window.location.hostname.includes("dev")) && <NotProductionWarning/>
            }
            <BetaBanner/>
            <Header title={"Generate UACs"}/>
            <div style={divStyle} className="page__container container">
                <main id="main-content" className="page__main">
                    <Switch>
                        <Route path="/upload">
                            <UploadSamplePage/>
                        </Route>
                        <Route path="/import">
                            <ImportUacPage/>
                        </Route>
                        <Route path="/">
                            <ul className="list list--bare list--inline u-mt-m">
                                <li className="list__item">
                                    <Link to="/upload" id="upload-sample-link">
                                        Upload sample
                                    </Link>
                                </li>
                                <li className="list__item">
                                    <Link to="/import" id="import-uacs-link">
                                        Upload used UACs
                                    </Link>
                                </li>
                            </ul>
                            <h2 className="u-mt-m">Previously uploaded questionnaire samples</h2>
                            <InstrumentListPage />
                        </Route>
                    </Switch>
                </main>
            </div>
            <Footer/>
        </>
    );
}

export default App;
