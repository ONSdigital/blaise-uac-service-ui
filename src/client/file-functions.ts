import axios from "axios";
import {Datas} from "react-csv-downloader/dist/esm/lib/csv";

export async function importUacsFromFile(file: File | undefined): Promise<number> {
    if (file === undefined) {
        throw new Error("file was not supplied");
    }

    const data = new FormData();
    data.append("file", file);

    const config = {
        headers: { "Content-Type": "multipart/form-data" },
        maxContentLength: Infinity,
        maxBodyLength: Infinity,
    };

    return axios.post("/api/v1/uac/import", data, config)
    .then((response) => {
        console.log("import-file-function - true");
        console.log(response.data);
        return response.data.uacs_imported;
    })
    .catch((error) => {
        console.log("import-file-function - false");
        console.error(`Something went wrong importing uacs ${error}`);
        throw error;
    });
}

export async function generateUacCodesForSampleFile(instrumentName: string | undefined, file: File | undefined): Promise<boolean> {
    if (instrumentName === undefined) {
        throw new Error("Questionnaire name was not supplied");
    }

    if (file === undefined) {
        throw new Error("file was not supplied");
    }

    const data = new FormData();
    data.append("fileName", getFileName(instrumentName));
    data.append("file", file);

    const config = {headers: {"Content-Type": "multipart/form-data"}};

    return axios.post(`/api/v1/instrument/${instrumentName}/uac/sample`, data, config)
        .then(() => {
            console.log("file-functions - true");
            return true;
        })
        .catch((error) => {
            console.log("file-functions - false");
            console.error(`Something went wrong in calling generate UAC endpoint ${error}`);
            return false;
        });

    console.log("file-functions - balls");
    return false;
}

export async function getSampleFileWithUacCodes(instrumentName: string | undefined, fileName: string | undefined): Promise<Datas> {
    if (instrumentName === undefined) {
        throw new Error("Questionnaire name was not supplied");
    }

    if (fileName === undefined) {
        throw new Error("file name was not supplied");
    }

    const response = await axios.get(`/api/v1/instrument/${instrumentName}/uac/sample/${fileName}`);

    return response.data;
}

export async function sampleFileAlreadyExists(instrumentName: string | undefined): Promise<boolean> {
    if (instrumentName === undefined) {
        throw new Error("Questionnaire name was not supplied");
    }

    const fileName = getFileName(instrumentName);
    return axios.get(`/api/v1/file/${fileName}/exists`)
        .then((response) => response.data === true)
        .catch((error) => {
            console.error(error);
            return false;
        });
}

export async function getListOfInstrumentsWhichHaveExistingSampleFiles(): Promise<string[]> {
    const response = await axios.get("/api/v1/instruments");

    return response.data;
}

export function getFileName(instrumentName: string): string {
    return `${instrumentName}.csv`;
}
