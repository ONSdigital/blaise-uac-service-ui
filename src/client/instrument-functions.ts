import axios from "axios";

export async function getInstrumentsWithExistingUacCodes(): Promise<string[]> {
    const response = await axios.get("/api/v1/instruments2");
    console.log("List of instruments: ", response);

    return [];
}
