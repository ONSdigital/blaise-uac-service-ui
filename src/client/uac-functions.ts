import axios from "axios";

export async function getInstrumentsWithExistingUacCodes() {
    const response = await axios.get("/api/v1/uac");
    console.log("Uac meh response", response);

    return response;
}
