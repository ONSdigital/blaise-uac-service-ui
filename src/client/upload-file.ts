import axios from "axios";

export async function uploadFile(instrumentName: string | undefined, file: File | undefined) {
    if (instrumentName === undefined || instrumentName === null) {
        console.error("Instrument name not supplied");
        return null;
    }

    if (file === undefined || file === null) {
        console.error("File not supplied");
        return null;
    }

    const data = new FormData();
    data.append("instrumentName", instrumentName);
    data.append("file", file);

    const config = {headers: {"Content-Type": "multipart/form-data"}};

    await axios.post("/api/v1/upload", data, config);
}
