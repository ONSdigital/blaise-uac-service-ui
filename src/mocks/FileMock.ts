import {Readable} from "stream";

export function fileMock({name = "file.txt", size = 1024, type = "plain/txt"}) {
    const blob = new Blob(["a".repeat(size)], {type});

    return new File([blob], name);
};

export function multerFileMock(
    {
        filename = "file.txt",
        path = "/temp/path",
        size = 1024,
        mimetype = "plain/txt",
        buffer = new Buffer(""),
        destination = "destinationTest",
        fieldname = "fieldNameTest",
        stream = new Readable(),
        originalname = "originalNameTest",
        encoding = ""
    }): Express.Multer.File {

    return {filename, path, size, mimetype, buffer, destination, fieldname, stream, originalname, encoding};
}
