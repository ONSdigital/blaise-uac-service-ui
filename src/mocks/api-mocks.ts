import { Questionnaire } from "blaise-api-node-client";
import { InstrumentUacDetails } from "blaise-uac-service-node-client";
import { QuestionnaireWithDisabledUacs } from "../models/model";

export const instrumentNames: string[] = [
    "OPN2101A",
    "DST1345A",
    "DST5643A",
];

export const questionnaireList: Questionnaire[] = [{
    name: "OPN2101A",
    serverParkName: "gusty",
    installDate: "2021-01-15T14:41:29.4399898+00:00",
    status: "Active",
    dataRecordCount: 0,
    hasData: false,
    active: false,
    fieldPeriod: "January 2021",
    blaiseVersion: "5.9.9.2735"
}, {
    name: "OPN2007T",
    serverParkName: "gusty",
    installDate: "2021-01-15T15:18:40.1503617+00:00",
    status: "Active",
    dataRecordCount: 10,
    hasData: true,
    active: true,
    fieldPeriod: "July 2020",
    blaiseVersion: "5.9.9.2735"

}, {
    name: "OPN2004A",
    serverParkName: "gusty",
    installDate: "2021-01-15T15:26:43.4233454+00:00",
    status: "Failed",
    dataRecordCount: 0,
    hasData: false,
    active: false,
    fieldPeriod: "April 2020",
    blaiseVersion: "5.9.9.2735"

}];

export const questionnaireListForEnableUacsMock: Questionnaire[] = [{
    name: "LMS2209_EM1",
    serverParkName: "gusty",
    installDate: "2021-01-15T14:41:29.4399898+00:00",
    status: "Active",
    dataRecordCount: 0,
    hasData: false,
    active: false,
    fieldPeriod: "January 2021",
    blaiseVersion: "5.9.9.2735"
}, {
    name: "LMS2207_HO1",
    serverParkName: "gusty",
    installDate: "2021-01-15T15:18:40.1503617+00:00",
    status: "Active",
    dataRecordCount: 10,
    hasData: true,
    active: true,
    fieldPeriod: "July 2020",
    blaiseVersion: "5.9.9.2735"

}];

export const disabledUacCodesForQuestionnaireMock: InstrumentUacDetails = {
    "803920": {
        "instrument_name": "lms2209_em1",
        "case_id": "803920",
        "uac_chunks": { "uac1": "1002", "uac2": "2293", "uac3": "8976" },
        "full_uac": "100222938976",
        "disabled": "true"
    },
    "804138": {
        "instrument_name": "lms2209_em1",
        "case_id": "804138",
        "uac_chunks": { "uac1": "1002", "uac2": "6087", "uac3": "6564" },
        "full_uac": "100260876564",
        "disabled": "true"
    },
    "907195": {
        "instrument_name": "lms2209_em1",
        "case_id": "907195",
        "uac_chunks": { "uac1": "1004", "uac2": "6119", "uac3": "7282" },
        "full_uac": "100461197282",
        "disabled": "true"
    }
};

export const questionnaireWithDisabledUacsMock: QuestionnaireWithDisabledUacs =
{
    "questionnaireName": "LMS2209_EM1",
    "disabledUacs": [
        {
            "case_id": "803920",
            "uac": "100222938976"
        },
        {
            "case_id": "804138",
            "uac": "100260876564"
        },
        {
            "case_id": "907195",
            "uac": "100461197282"
        },

    ]
};

export const questionnaireWithOneDisabledUacMock: QuestionnaireWithDisabledUacs =
{
    "questionnaireName": "LMS2209_EM1",
    "disabledUacs": [
        {
            "case_id": "803920",
            "uac": "100222938976"
        }

    ]
};