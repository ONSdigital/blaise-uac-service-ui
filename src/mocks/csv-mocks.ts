import { InstrumentUacDetailsByCaseId } from "blaise-uac-service-node-client";
import { Datas } from "react-csv-downloader/dist/esm/lib/csv";

export const validUACImportCsv = `Full_UAC
123412341234
432143214321
678967896789
987698769876
`;

export const validSampleCsv = `serial_number,Name,Phone Number,Email
100000001,Homer Simpson,5551234422,homer@springfield.com
100000002,Seymour Skinner,1235663322,a@b.c
100000003,Bart Simpson,2675465026,bart@spring.field
`;

export const validSampleCsvWithExistingUacColumns = `serial_number,Name,UAC1,UAC2,UAC3,Phone Number,Email
100000001,Homer Simpson,,,,5551234422,homer@springfield.com
100000002,Seymour Skinner,,,,1235663322,a@b.c
100000003,Bart Simpson,,,,2675465026,bart@spring.field
`;

export const validSampleCsvWithExistingLowercaseUacColumns = `serial_number,Name,uac1,uac2,uac3,Phone Number,Email
100000001,Homer Simpson,,,,5551234422,homer@springfield.com
100000002,Seymour Skinner,,,,1235663322,a@b.c
100000003,Bart Simpson,,,,2675465026,bart@spring.field
`;

export const validSampleCsvWithExistingMixedCaseUacColumns = `serial_number,Name,UaC1,UaC2,UaC3,Phone Number,Email
100000001,Homer Simpson,,,,5551234422,homer@springfield.com
100000002,Seymour Skinner,,,,1235663322,a@b.c
100000003,Bart Simpson,,,,2675465026,bart@spring.field
`;

export const validSampleCsvWithExistingUacEntries = `serial_number,Name,UAC1,UAC2,UAC3,Phone Number,Email
100000001,Homer Simpson,3454,5453,5353,5551234422,homer@springfield.com
100000002,Seymour Skinner,8786,2213,3343,1235663322,a@b.c
100000003,Bart Simpson,2242,5543,7676,2675465026,bart@spring.field
`;

export const invalidSampleCsv = `serial_number,Phone Number,Email
100000001,Homer Simpson,5551234422,homer@springfield.com
100000002,Seymour Skinner,1235663322,a@b.c
100000003,Bart Simpson,2675465026,bart@spring.field
`;


export const matchedInstrumentUacDetails: InstrumentUacDetailsByCaseId = {
    "100000001": {
        instrument_name: "dst2106a",
        case_id: "100000001",
        uac_chunks: {
            uac1: "0009",
            uac2: "7565",
            uac3: "3827"
        },
        full_uac: "000975653827"
    },
    "100000002": {
        instrument_name: "dst2106a",
        case_id: "100000002",
        uac_chunks: {
            uac1: "3453",
            uac2: "6545",
            uac3: "4564"
        },
        full_uac: "345365454564"
    },
    "100000003": {
        instrument_name: "dst2106a",
        case_id: "100000003",
        uac_chunks: {
            uac1: "9789",
            uac2: "7578",
            uac3: "5367"
        },
        full_uac: "978975785367"
    }
};

export const matchedInstrumentUac16Details: InstrumentUacDetailsByCaseId = {
    "100000001": {
        instrument_name: "dst2106a",
        case_id: "100000001",
        uac_chunks: {
            uac1: "0009",
            uac2: "7565",
            uac3: "3827",
            uac4: "7512"
        },
        full_uac: "0009756538277512"
    },
    "100000002": {
        instrument_name: "dst2106a",
        case_id: "100000002",
        uac_chunks: {
            uac1: "3453",
            uac2: "6545",
            uac3: "4564",
            uac4: "3213"
        },
        full_uac: "3453654545643213"
    },
    "100000003": {
        instrument_name: "dst2106a",
        case_id: "100000003",
        uac_chunks: {
            uac1: "9789",
            uac2: "7578",
            uac3: "5367",
            uac4: "8765"
        },
        full_uac: "9789757853678765"
    }
};

export const unMatchedInstrumentUacDetails: InstrumentUacDetailsByCaseId = {
    "100000997": {
        instrument_name: "dst2106a",
        case_id: "100000997",
        uac_chunks: {
            uac1: "0009",
            uac2: "7565",
            uac3: "3827"
        },
        full_uac: "000975653827"
    },
    "100000998": {
        instrument_name: "dst2106a",
        case_id: "100000998",
        uac_chunks: {
            uac1: "3453",
            uac2: "6545",
            uac3: "4564"
        },
        full_uac: "345365454564"
    },
    "100000999": {
        instrument_name: "dst2106a",
        case_id: "100000999",
        uac_chunks: {
            uac1: "9789",
            uac2: "7578",
            uac3: "5367"
        },
        full_uac: "978975785367"
    }
};

export const partialMatchedInstrumentUacDetails: InstrumentUacDetailsByCaseId = {
    "100000002": {
        instrument_name: "dst2106a",
        case_id: "100000002",
        uac_chunks: {
            uac1: "3453",
            uac2: "6545",
            uac3: "4564"
        },
        full_uac: "345365454564"
    }
};

export const emptyInstrumentUacDetails: InstrumentUacDetailsByCaseId = {};

export const validSampleFileWithUacArrayResponse: any[] = [
    {
        serial_number: "100000001",
        name: "Homer Simpson",
        phone: "5551234422",
        email: "homer@springfield.com",
        UAC1: "0009",
        UAC2: "7565",
        UAC3: "3827"
    },
    {
        serial_number: "100000002",
        name: "Seymour Skinner",
        phone: "1235663322",
        email: "a@b.c",
        UAC1: "3453",
        UAC2: "6545",
        UAC3: "4564"
    }
];

export const validSampleFileWithUacDatasResponse: Datas = [
    {
        serial_number: "100000001",
        name: "Homer Simpson",
        phone: "5551234422",
        email: "homer@springfield.com",
        UAC1: "0009",
        UAC2: "7565",
        UAC3: "3827"
    },
    {
        serial_number: "100000002",
        name: "Seymour Skinner",
        phone: "1235663322",
        email: "a@b.c",
        UAC1: "3453",
        UAC2: "6545",
        UAC3: "4564"
    }
];
