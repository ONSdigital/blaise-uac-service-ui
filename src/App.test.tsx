/**
 * @jest-environment jsdom
 */

import React from "react";
import { render, waitFor, act } from "@testing-library/react";
import App from "./App";
import { MemoryRouter as Router } from "react-router";
import "@testing-library/jest-dom";
import { instrumentNames } from "./mocks/api-mocks";
import { AuthManager } from "blaise-login-react-client";

jest.mock("blaise-login-react-client");
AuthManager.prototype.loggedIn = jest.fn().mockImplementation(() => {
    return Promise.resolve(true);
});

jest.mock("./client/file-functions");
import { getListOfInstrumentsWhichHaveExistingSampleFiles } from "./client/file-functions";

const mockIsProduction = jest.fn();
jest.mock("./client/env", () => ({
    isProduction: () => mockIsProduction()
}));

const getListOfInstrumentsWhichHaveExistingSampleFilesMock = getListOfInstrumentsWhichHaveExistingSampleFiles as jest.Mock<Promise<string[]>>;

describe("React homepage", () => {
    beforeAll(() => {
        getListOfInstrumentsWhichHaveExistingSampleFilesMock.mockImplementation(() => Promise.resolve(instrumentNames));
    });

    beforeEach(() => {
        jest.clearAllMocks();
        jest.resetModules();
        mockIsProduction.mockReturnValue(false);
    });

    it("App page matches snapshot", async () => {
        const wrapper = render(<App />, { wrapper: Router });
        await waitFor(() => {
            expect(wrapper).toMatchSnapshot();
        });
    });

    // it("should render correctly", async () => {
    //     const { queryByText } = render(<App />, { wrapper: Router });

    //     await act(async () => await waitFor(() => {
    //         expect(queryByText(/This environment is not a production environment. Do not upload any live data to this service./i)).toBeInTheDocument();
    //         expect(queryByText(/Previously uploaded questionnaire samples/i)).toBeInTheDocument();
    //         instrumentNames.forEach((instrumentName) => {
    //             expect(queryByText(instrumentName)).toBeInTheDocument();
    //         });
    //     }));
    // });

    it("view instrument page matches Snapshot in production", async () => {
        mockIsProduction.mockReturnValue(true);
        const { queryByText } = render(<App />, { wrapper: Router });
        await waitFor(() => {
            expect(queryByText(/This environment is not a production environment. Do not upload any live data to this service./i)).not.toBeInTheDocument();
        });
    });
});