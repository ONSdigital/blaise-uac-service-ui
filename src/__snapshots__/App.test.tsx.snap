// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React homepage App page matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="phase-banner"
      >
        <div
          class="container "
        >
          <div
            class="grid grid--flex grid--gutterless grid--vertical-center grid--no-wrap"
          >
            <div
              class="grid__col col-auto u-flex-no-grow"
            >
              <h3
                class="phase-banner__badge"
              >
                BETA
              </h3>
            </div>
            <div
              class="grid__col col-auto u-flex-shrink"
            >
              <p
                class="phase-banner__desc u-fs-s u-mb-no"
              >
                This is a new service – your 
                <a
                  href="https://ons.service-now.com/"
                >
                  feedback
                </a>
                 will help us improve it.
              </p>
            </div>
          </div>
        </div>
      </div>
      <header
        class="header header--internal"
      >
        <div
          class="header__top"
          role="banner"
        >
          <div
            class="container"
          >
            <div
              class="header__grid-top grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap "
            >
              <div
                class="grid__col col-auto"
              >
                <a
                  class="header__logo-link"
                  href="/"
                >
                  <picture>
                    <img
                      alt="Office for National Statistics logo"
                      class="header__logo"
                      src="https://cdn.ons.gov.uk/sdc/design-system/31.4.0/img/ons-logo-neg-en.svg"
                    />
                  </picture>
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="header__main"
        >
          <div
            class="container"
          >
            <div
              class="grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap"
            >
              <div
                class="grid__col col-auto u-flex-shrink"
              >
                <div
                  class="header__title"
                >
                  Generate UACs
                </div>
              </div>
            </div>
          </div>
        </div>
      </header>
      <div
        class="page__container container"
        style="min-height: calc(67vh);"
      >
        <main
          class="page__main"
          id="main-content"
        >
          <ul
            class="list list--bare list--inline u-mt-m"
          >
            <li
              class="list__item"
            >
              <a
                href="/upload"
                id="upload-sample-link"
              >
                Upload sample
              </a>
            </li>
            <li
              class="list__item"
            >
              <a
                href="/import"
                id="import-uacs-link"
              >
                Upload used UACs
              </a>
            </li>
          </ul>
          <h2
            class="u-mt-m"
          >
            Previously uploaded questionnaire samples
          </h2>
          <div>
            <div
              class="u-mt-m"
            >
              <div
                class="panel panel--info panel--no-title panel--spacious u-mt-m"
              >
                <div
                  class="panel__body "
                />
              </div>
            </div>
          </div>
        </main>
      </div>
      <footer
        class="footer"
      >
        <div
          class="footer__body"
          data-analytics="footer"
        >
          <div
            class="container "
          >
            <div
              class="grid grid--flex grid--between"
            >
              <div
                class="grid__col"
              >
                <div
                  class="u-mt-m u-mb-s"
                >
                  <img
                    alt="Office for National Statistics"
                    src="https://ons-design-system.netlify.app/img/ons-logo-black-en.svg"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <div
      class="phase-banner"
    >
      <div
        class="container "
      >
        <div
          class="grid grid--flex grid--gutterless grid--vertical-center grid--no-wrap"
        >
          <div
            class="grid__col col-auto u-flex-no-grow"
          >
            <h3
              class="phase-banner__badge"
            >
              BETA
            </h3>
          </div>
          <div
            class="grid__col col-auto u-flex-shrink"
          >
            <p
              class="phase-banner__desc u-fs-s u-mb-no"
            >
              This is a new service – your 
              <a
                href="https://ons.service-now.com/"
              >
                feedback
              </a>
               will help us improve it.
            </p>
          </div>
        </div>
      </div>
    </div>
    <header
      class="header header--internal"
    >
      <div
        class="header__top"
        role="banner"
      >
        <div
          class="container"
        >
          <div
            class="header__grid-top grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap "
          >
            <div
              class="grid__col col-auto"
            >
              <a
                class="header__logo-link"
                href="/"
              >
                <picture>
                  <img
                    alt="Office for National Statistics logo"
                    class="header__logo"
                    src="https://cdn.ons.gov.uk/sdc/design-system/31.4.0/img/ons-logo-neg-en.svg"
                  />
                </picture>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="header__main"
      >
        <div
          class="container"
        >
          <div
            class="grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap"
          >
            <div
              class="grid__col col-auto u-flex-shrink"
            >
              <div
                class="header__title"
              >
                Generate UACs
              </div>
            </div>
          </div>
        </div>
      </div>
    </header>
    <div
      class="page__container container"
      style="min-height: calc(67vh);"
    >
      <main
        class="page__main"
        id="main-content"
      >
        <ul
          class="list list--bare list--inline u-mt-m"
        >
          <li
            class="list__item"
          >
            <a
              href="/upload"
              id="upload-sample-link"
            >
              Upload sample
            </a>
          </li>
          <li
            class="list__item"
          >
            <a
              href="/import"
              id="import-uacs-link"
            >
              Upload used UACs
            </a>
          </li>
        </ul>
        <h2
          class="u-mt-m"
        >
          Previously uploaded questionnaire samples
        </h2>
        <div>
          <div
            class="u-mt-m"
          >
            <div
              class="panel panel--info panel--no-title panel--spacious u-mt-m"
            >
              <div
                class="panel__body "
              />
            </div>
          </div>
        </div>
      </main>
    </div>
    <footer
      class="footer"
    >
      <div
        class="footer__body"
        data-analytics="footer"
      >
        <div
          class="container "
        >
          <div
            class="grid grid--flex grid--between"
          >
            <div
              class="grid__col"
            >
              <div
                class="u-mt-m u-mb-s"
              >
                <img
                  alt="Office for National Statistics"
                  src="https://ons-design-system.netlify.app/img/ons-logo-black-en.svg"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
